opentelemetry-instrumentation/tests/test_semconv.py

```diff
@@ -0,0 +1,222 @@
+                _OpenTelemetryStabilitySignalType.DATABASE
+            ),
+            _StabilityMode.DATABASE_DUP,
+        )
+        self.assertEqual(
+            _OpenTelemetrySemanticConventionStability._get_opentelemetry_stability_opt_in_mode(
+                _OpenTelemetryStabilitySignalType.HTTP
+            ),
+            _StabilityMode.HTTP_DUP,
+        )
+
+
+class TestOpenTelemetrySemConvStabilityHTTP(TestCase):
+    def test_set_status_for_non_http_code_with_recording_span(self):
+        span = Mock()
+        span.is_recording.return_value = True
+        metric_attributes = {}
+        _set_status(
+            span,
+            metric_attributes,
+            -1,
+            "Exception",
+            sem_conv_opt_in_mode=_StabilityMode.DEFAULT,
+        )
+
+        self.assertIsNone(metric_attributes.get("error.type"))
+        span.set_attribute.assert_not_called()
+        status_call = span.set_status.call_args[0][0]
+        self.assertEqual(status_call.status_code, StatusCode.ERROR)
+        self.assertEqual(
+            status_call.description, "Non-integer HTTP status: " + "Exception"
+        )
+
+    def test_status_code_http_default(self):
+        span = Mock()
+        metrics_attributes = {}
+        _set_status(
+            span=span,
+            metrics_attributes=metrics_attributes,
+            status_code=404,
+            status_code_str="404",
+            server_span=True,
+            sem_conv_opt_in_mode=_StabilityMode.DEFAULT,
+        )
+        # Verify only old conventions are emitted
+        span.set_attribute.assert_called_with("http.status_code", 404)
+        self.assertIn("http.status_code", metrics_attributes)
+        self.assertNotIn("http.response_status_code", metrics_attributes)
+
+    def test_status_code_http_stable(self):
+        span = Mock()
+        metrics_attributes = {}
+        _set_status(
+            span=span,
+            metrics_attributes=metrics_attributes,
+            status_code=200,
+            status_code_str="200",
+            server_span=True,
+            sem_conv_opt_in_mode=_StabilityMode.HTTP,
+        )
+        # Verify only new conventions are emitted
+        span.set_attribute.assert_called_with("http.response.status_code", 200)
+        self.assertIn("http.response.status_code", metrics_attributes)
+        self.assertNotIn("http.status_code", metrics_attributes)
+
+    def test_status_code_http_dup(self):
+        span = Mock()
+        metrics_attributes = {}
+        _set_status(
+            span=span,
+            metrics_attributes=metrics_attributes,
+            status_code=500,
+            status_code_str="500",
+            server_span=True,
+            sem_conv_opt_in_mode=_StabilityMode.HTTP_DUP,
+        )
+        # Verify both old and new conventions are emitted
+        span.set_attribute.assert_any_call("http.status_code", 500)
+        span.set_attribute.assert_any_call("http.response.status_code", 500)
```

Comment:
These tests are helpful, thank you!

It would be great to see a similar `database/dup` implementation but maybe that's better for a different set of PRs? For example, if `database/dup`then I'd expect `set_attribute` of both [db.statement](https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-semantic-conventions/src/opentelemetry/semconv/trace/__init__.py#L423-L426) (old) and db.query.text/db.query.summary (new). But I don't think we've updated [semconv.attributes](https://github.com/open-telemetry/opentelemetry-python/tree/main/opentelemetry-semantic-conventions/src/opentelemetry/semconv/attributes) with this yet

Reply:
I think we can do that as per as we migrate instrumentation to use the stability opt_in. Wdyt? Right now is hard to write tests because I didn't read all the changes 😅

Reply:
That works! This is good for the contrib config part